source/fundamentals/data-formats/document-data-format-record.txt

```diff
@@ -0,0 +1,141 @@
+.. _fundamentals-records:
+
+=============================
+Document Data Format: Records
+=============================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to store and retrieve data in the {+driver-long+}
+using **Java records**. Java records are a type of Java class often used to
+model data and separate business logic from data representation.
+
+.. tip::
+
+   You can declare Java records in Java 14 or later. Learn more about the
+   functionality and restrictions of records from `Java 14 Language Updates: Records <https://docs.oracle.com/en/java/javase/14/language/records.html>`__.
+
+   If you are using an earlier version of Java, you can use plain old Java
+   objects instead. See the :ref:`<fundamentals-pojos>` guide for
+   implementation details.
+
+The following sections show how you can perform the following tasks:
+
+- Configure the driver to serialize and deserialize a record.
+- Read data from MongoDB into records.
+- Write data from records into MongoDB.
+
+.. _fundamentals-example-record:
+
+Example Record
+--------------
+
+The code examples in this guide reference the following sample record, which
+describes a data storage device:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/records/DataStorageRecord.java
+   :language: java
+   :start-after: start dataStorageRecord
+   :end-before: end dataStorageRecord
+
+Serialize and Deserialize a Record
+----------------------------------
+
+To set up the driver to serialize and deserialize a record, you must provide
```

Comment:
This is all true, but unlike for POJOs, the default codec registry includes the Jep395RecordCodecProvider, so the application doesn't have to configure the registry at all in order to serialize/deserialize a Java record.